### ISSUE TITLE
Flaky E2E: See and click notification for the comment left by commentingUser

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#69857]
### Spec file

specs/infrastructure/notification__actions.ts: Notifications

### TeamCity ID

8957836

### Logs

````
page.click: Timeout 30000ms exceeded.
=========================== logs ===========================
waiting for selector ".wpnc__comment:has-text("12 Large Whales Beg Softly notification-actions-spec")"
============================================================
at NotificationsComponent.clickNotification (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/packages/calypso-e2e/src/lib/components/notifications-component.ts:35:19)
    at Object.<anonymous> (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/infrastructure/notification__actions.ts:61:33)
    at Promise.then.completed (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:391:28)
    at callAsyncCircusFn (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:218:40)
    at _runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:60:9)
    at _runTestsForDescribeBlock (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:60:9)
    at run (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
    at jestAdapter (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:82:19)
    at runTestInternal (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:389:16)
    at runTest (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:475:34)
    at Object.worker (/home/teamcity-0/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/testWorker.js:133:12)
```

**Comment by worldomonation** [comment]
@john-legg We seem to have an approximately 10% failure rate on this step.

**Comment by john-legg** [comment]
@worldomonation I looked into the failures via the TeamCity Artifacts tab (thanks for telling me about that). It appears that the notification panel doesn't open when the home page takes longer than usual to load (on both desktop and mobile). 

My only idea is to check for other elements of the page to load (e.g. the "My Home" header) before opening the notifications panel. Unfortunately, I can't replicate it locally, so it's been hard to know if that will work or if it's even the issue.

**question: blocking**
Is there a more elegant way to handle the longer loading issue? Also, what's the best way to replicate it to ensure that my fix is actually helping?

**Comment by worldomonation** [comment]
> My only idea is to check for other elements of the page to load (e.g. the "My Home" header) before opening the notifications panel. Unfortunately, I can't replicate it locally, so it's been hard to know if that will work or if it's even the issue.

Just to double check, did you also take a look at the trace files? They are greatly useful and in many cases eliminate having to run the spec locally repeatedly.

> **question: blocking** Is there a more elegant way to handle the longer loading issue? Also, what's the best way to replicate it to ensure that my fix is actually helping?

I'll answer this in two parts.

> Is there a more elegant way to handle the longer loading issue?

There's always the `networkidle` call to force a wait until all network activity ceases for more than 500ms, but that's typically overkill. With that said, it might be necessary in some cases and we haven't been shy about deploying it in the past, just that we need to make absolutely sure that causing a wait for possibly 10 seconds is required.

Also, keep in mind Karen's investigation into https://github.com/Automattic/wp-calypso/issues/69294 - we may end up removing the default `networkidle` call in the account authentication flow, so you might have to make a targeted `networkidle` call to cover this case.

> Also, what's the best way to replicate it to ensure that my fix is actually helping?

Are you familiar with browser developer tools? You can throttle network calls there. It might be helpful in replicating the issue.

**Comment by john-legg** [comment]
I had a great chat with @worldomonation earlier where he showed me how to use the [The Playwright Trace Viewer](https://playwright.dev/docs/trace-viewer#viewing-the-trace) to investigate this issue further. It's **super** powerful! 

While briefly going through the trace together, a number of potential solutions came up:
- Update `pageSettled()` in `navbar-component.ts` to use methods more in-line with Playwright's current suggestions.
- Wait for some other object on the `navbarComponent` to load first (less ideal).
- Add a general wait to the test (least ideal).

In terms of replicating the failure, I've tried using the Chrome developer tools to throttle the network during debugging (download 376 kb/s, latency 2000 ms), but I still haven't run into it yet. Edwin said it may not be throttled enough, so I'll experiment with it some more.

In general, this issue might be why the test became quarantined in the first place 😓 I will continue to look into it with the new techniques Edwin showed me. Thanks again!

**Comment by worldomonation** [comment]
Hopefully addressed by #70348. Let's monitor for a bit.
